### DISCUSSION TITLE
Template language for PL/pgSQL

### DISCUSSION THREAD

**Discussion by LyMc** [supabase/supabase#3502] [Questions]
I'm looking for a solution to populate template strings with values from a row.
For a string that contains a variable `"Hello { user.name }"` I want to populate it with the data. Resulting in `"Hello LyMc"`.
For template syntax, it can be anything, like `{ user.name }`, `${ user.name }`, `{{ user.name }}`...

I need to do it in a trigger.
A template string can contain zero or more variables. All have to be populated.

What options do I have?

**Comment by GaryAustin1** [comment]
Not positive what you are asking for, but if you have a trigger function and want to get row data for the table the trigger was on you use new.col or old.col.  Old is for existing data, and New is data to be inserted or updated.

edit:  If you asking about how to format the actual string, I used concatenation the time I needed it but there is Format function: https://www.postgresqltutorial.com/postgresql-format/

**Reply by LyMc** [reply]
Yes, the question is about formatting the string. The `format` function can help, but I still need to replace templates with values, because a value can be anything from a list of ~10-20 values. I expect to provide user data and data about the changed row.
Eg. When a new post was published, I want to send a notification to the subscribed users about it. For this, I can set a trigger for inserts on the posts table, and there send notifications. I want that these notifications to contain data about the new post (or the user itself), so a configurable template will work best there. I just can't figure out how to replace templates with actual data.
